src/lib/rc/srxl.hpp

```diff
@@ -0,0 +1,184 @@
+/****************************************************************************
+ *
+ *	Copyright (c) 2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *	notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *	notice, this list of conditions and the following disclaimer in
+ *	the documentation and/or other materials provided with the
+ *	distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *	used to endorse or promote products derived from this software
+ *	without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file srxl.h
+ *
+ * RC protocol definition for Spektrum SRXL
+ *
+ * @author Kurt Kiefer <kekiefer@gmail.com>
+ * @author Daniel Williams <equipoise@gmail.com>
+ */
+
+#pragma once
+
+// "Cannot find the standard library!?"
+// #include <array>
+
+#include <cstdint>
+#include <px4_platform_common/defines.h>
+// #include <px4_platform_common/px4_config.h>
+
+class SRXLCodec
+{
+
+// ====== ====== ====== ====== Public API ====== ====== ====== ======
```

Comment:
Nitpick, I would prefer skipping things like this to stay consist across the code base.

Reply:
Specifically referring to the horizontal-rule section header? 

Is there an equivalent pattern, with different syntax you guys would prefer, or just remove it entirely?

Reply:
Okay, removed as requested.